implementations/elixir/ockam/ockam_hub/lib/hub/service/token_lease_manager.ex

```diff
@@ -1,130 +1,242 @@
 defmodule Ockam.TokenLeaseManager do
   @moduledoc false
-  use GenServer
+  use Ockam.Worker
 
   alias Ockam.TokenLeaseManager.Lease
+  alias Ockam.Node
 
-  @default_cloud_service :influxdb
-  @default_storate_service :storage
+  require Logger
 
-  def start_link(options) do
-    GenServer.start_link(__MODULE__, options, [])
+  @impl true
+  def setup(options, state) do
+    {:ok, Map.merge(state, initialization(options))}
   end
 
-  @impl true
-  def init(options) do
-    cloud_service_name =
-      case Keyword.fetch(options, :cloud_service) do
-        {:ok, cloud_service_name} -> cloud_service_name
-        _other -> @default_cloud_service
-      end
+  defp initialization(_options) do
+    token_manager_opts = Application.get_env(:ockam_hub, :token_manager)
+    cloud_service_name = token_manager_opts[:cloud_service]
+    storage_service_name = token_manager_opts[:storage_service]
+
+    with {:ok, cloud_service} <- get_cloud_service(cloud_service_name),
+      {:ok, cloud_service_address} <- cloud_service.get_address(),
+         {:ok, storage_service} <-
+           get_storage_service(storage_service_name, cloud_service_name, cloud_service_address) do
+      # Initializing all saved ttls
+      send(self(), :set_all_ttl)
+
+      %{
+        cloud_service: cloud_service,
+        storage_service: storage_service
+      }
+    else
```

Comment:
Should we move that branch to `setup`? I suspect it would crash on this error otherwise.

Reply:
sure